app/lib/account/models.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:gcloud/db.dart' as db;
+
+/// User data model.
+@db.Kind(name: 'User', idType: db.IdType.String)
+class User extends db.ExpandoModel {
+  String get userId => id as String;
+
+  @db.StringProperty()
+  String authProvider;
+
+  @db.StringProperty()
+  String authUserId;
+
+  @db.StringProperty()
+  String email;
+
+  @db.StringListProperty()
+  List<String> emails = <String>[];
+
+  @db.BoolProperty()
+  bool hideEmail = false;
+
+  @db.DateTimeProperty()
+  DateTime created;
+
+  @db.DateTimeProperty()
+  DateTime updated;
+
+  @db.DateTimeProperty()
+  DateTime lastAccess;
+
+  bool updateEmail(String email) {
+    bool changed = false;
+
+    if (this.email != email) {
+      this.email = email;
+      changed = true;
+    }
+    if (emails == null || !emails.contains(email)) {
+      this.email = email;
+      emails ??= <String>[];
```

Comment:
Is this what happens for existing users?
I have not worked with these data classes before. Is there some way to ensure it is not null from a constructor? It seems a bit brittle that we always have to care about the null case.

Alternatively you could reorder the logic:
```dart
emails ??= <String>[];
if (!emails.contains(email)) {\
  ...
}
```
A tiny bit of overhead, but I think easier to read.

Reply:
It was me being too paranoid. Having checked the `ListProperty` sources, it always returns a non-empty String on decoding the database content.